hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDWriteClient.java

```diff
@@ -389,15 +389,16 @@ private void completeClustering(HoodieReplaceCommitMetadata metadata, JavaRDD<Wr
     writeTableMetadata(table, metadata, new HoodieInstant(HoodieInstant.State.INFLIGHT, HoodieTimeline.REPLACE_COMMIT_ACTION, clusteringCommitTime));
     finalizeWrite(table, clusteringCommitTime, writeStats);
     try {
-      // try to save statistics info to hudi
-      if (config.isDataSkippingEnabled() && config.isLayoutOptimizationEnabled() && !config.getClusteringSortColumns().isEmpty()) {
-        table.updateStatistics(context, writeStats, clusteringCommitTime, true);
+      // Update outstanding metadata indexes
+      if (config.isLayoutOptimizationEnabled()
```

Comment:
I had a similar comment before. this. is a good change

Reply:
If you really want to modify it in this way, pls remove layout_ OPTIMIZE_ DATA_ SKIPPING_ Enable this configuration item.
This configuration item was introduced for fear of unstable data skipping， let's remove  this config.

by the way, it is suggested to directly remove this judgment condition. In this way, the cluster + sort operation can also generate indexes for query

Reply:
For now, I think this is okay to be decoupled? I see what you are saying though. I will try adding a new config to just create data skipping indexes decoupled from space curves or linear sorting (time is the only issue)